---
source: crates/oxc_linter/src/tester.rs
expression: no_useless_promise_resolve_reject
---
  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:4:28]
 3 â”‚                 if (foo > 4) {
 4 â”‚                     return Promise.reject(new Error('ðŸ¤ª'));
   Â·                            â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 5 â”‚                 }
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:6:24]
 5 â”‚                 }
 6 â”‚                 return Promise.resolve(result);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 7 â”‚             };
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.resolve(bar);
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:24]
 2 â”‚             async () => {
 3 â”‚                 return Promise.resolve(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             };
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:24]
 2 â”‚             async function foo() {
 3 â”‚                 return Promise.resolve(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             }
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚             (async function() {
 3 â”‚                 return Promise.resolve(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             });
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚             async function * foo() {
 3 â”‚                 return Promise.resolve(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             }
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:15]
 2 â”‚             (async function*() {
 3 â”‚                 return Promise.resolve(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             });
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.reject(bar);
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚             async () => {
 3 â”‚                 return Promise.reject(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             };
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚             async function foo() {
 3 â”‚                 return Promise.reject(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             }   
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚             (async function() {
 3 â”‚                 return Promise.reject(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             });
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚             async function * foo() {
 3 â”‚                 return Promise.reject(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             }
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚             (async function*() {
 3 â”‚                 return Promise.reject(bar);
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             });
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `yield value` over `yield Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚                 async function * foo() {
 3 â”‚                     yield Promise.resolve(bar);
   Â·                           â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚                 }
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `yield value` over `yield Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚                 (async function * () {
 3 â”‚                     yield Promise.resolve(bar);
   Â·                           â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚                 });
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `yield Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚                 async function * foo() {
 3 â”‚                     yield Promise.reject(bar);
   Â·                           â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚                 }
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `yield Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚                 (async function * () {
 3 â”‚                     yield Promise.reject(bar);
   Â·                           â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚                 });
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.resolve();
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:13]
 2 â”‚                 async function foo() {
 3 â”‚                     return Promise.resolve();
   Â·                            â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚                 }
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.reject();
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:13]
 2 â”‚                 async function foo() {
 3 â”‚                     return Promise.reject();
   Â·                            â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚                 }
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `yield value` over `yield Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚                 async function * foo() {
 3 â”‚                     yield Promise.resolve();
   Â·                           â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚                 }
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.resolve(bar, baz);
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.reject(bar, baz);
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `yield value` over `yield Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:12]
 2 â”‚                 async function * foo() {
 3 â”‚                     yield Promise.resolve((bar, baz));
   Â·                           â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚                 }
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.resolve((bar, baz))
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.resolve({})
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:4:14]
 3 â”‚                     try {
 4 â”‚                         return Promise.resolve(1);
   Â·                                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 5 â”‚                     } catch {}
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:4:14]
 3 â”‚                     try {
 4 â”‚                         return Promise.reject(1);
   Â·                                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 5 â”‚                     } catch {}
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.resolve(...bar);
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:13]
 1 â”‚ async () => Promise.reject(...bar);
   Â·             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  Ã— Invalid Character `
  â”‚ `
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:2]
 1 â”‚ #
   Â·  â”€
 2 â”‚         async function * foo() {
 3 â”‚             const baz = yield Promise.resolve(bar);
   â•°â”€â”€â”€â”€

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `yield Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:31]
 2 â”‚         async function * foo() {
 3 â”‚             const baz = yield Promise.reject(bar);
   Â·                               â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚         }
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:14]
 1 â”‚ async () => (Promise.resolve(bar));
   Â·              â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:14]
 1 â”‚ async () => (Promise.reject(bar));
   Â·              â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:15]
 1 â”‚ async () => ((Promise.reject(bar)));
   Â·               â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `yield Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:16]
 2 â”‚             async function * foo() {
 3 â”‚                 (yield Promise.reject(bar));
   Â·                        â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             }
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `yield Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:3:17]
 2 â”‚             async function * foo() {
 3 â”‚                 ((yield Promise.reject(bar)));
   Â·                         â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
 4 â”‚             }
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:20]
 1 â”‚ promise.then(() => Promise.resolve(bar))
   Â·                    â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:29]
 1 â”‚ promise.then(() => { return Promise.resolve(bar); })
   Â·                             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:26]
 1 â”‚ promise.then(async () => Promise.reject(bar))
   Â·                          â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:35]
 1 â”‚ promise.then(async () => { return Promise.reject(bar); })
   Â·                                   â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:21]
 1 â”‚ promise.catch(() => Promise.resolve(bar))
   Â·                     â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:30]
 1 â”‚ promise.catch(() => { return Promise.resolve(bar); })
   Â·                              â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:27]
 1 â”‚ promise.catch(async () => Promise.reject(bar))
   Â·                           â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:36]
 1 â”‚ promise.catch(async () => { return Promise.reject(bar); })
   Â·                                    â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:23]
 1 â”‚ promise.finally(() => Promise.resolve(bar))
   Â·                       â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:32]
 1 â”‚ promise.finally(() => { return Promise.resolve(bar); })
   Â·                                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:29]
 1 â”‚ promise.finally(async () => Promise.reject(bar))
   Â·                             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:38]
 1 â”‚ promise.finally(async () => { return Promise.reject(bar); })
   Â·                                      â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:30]
 1 â”‚ promise.then(() => {}, () => Promise.resolve(bar))
   Â·                              â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:20]
 1 â”‚ promise.then(() => Promise.resolve(bar), () => Promise.resolve(baz))
   Â·                    â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:48]
 1 â”‚ promise.then(() => Promise.resolve(bar), () => Promise.resolve(baz))
   Â·                                                â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `return value` over `return Promise.resolve(value)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:39]
 1 â”‚ promise.then(() => {}, () => { return Promise.resolve(bar); })
   Â·                                       â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the return value in `Promise.Resolve` is needlessly verbose. All return values in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:36]
 1 â”‚ promise.then(() => {}, async () => Promise.reject(bar))
   Â·                                    â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.

  âš  eslint-plugin-unicorn(no-useless-promise-resolve-reject): Prefer `throw error` over `return Promise.reject(error)`.
   â•­â”€[no_useless_promise_resolve_reject.tsx:1:45]
 1 â”‚ promise.then(() => {}, async () => { return Promise.reject(bar); })
   Â·                                             â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
   â•°â”€â”€â”€â”€
  help: Wrapping the error in `Promise.reject` is needlessly verbose. All errors thrown in async functions are already wrapped in a `Promise`.
